cpp/src/arrow/util/thread_pool.cc

```diff
@@ -168,9 +174,11 @@ Status ThreadPool::SetCapacity(int threads) {
   CollectFinishedWorkersUnlocked();
 
   state_->desired_capacity_ = threads;
-  int diff = static_cast<int>(threads - state_->workers_.size());
-  if (diff > 0) {
-    LaunchWorkersUnlocked(diff);
+  const int diff = threads - static_cast<int>(state_->workers_.size());
```

Comment:
Maybe easier to read?
```suggestion
  const int required = static_cast<int>(std::min(
    threads - state_->workers_.size(),
    state_->pending_tasks_.size()));
```